Functions/FirebaseFunctions/FIRFunctions.m

```diff
@@ -95,30 +95,57 @@ + (void)load {
   return @[ internalProvider ];
 }
 
+// A table mapping FIRApp instances to FIRStorage instances, used to cache FIRStorage instances
+// on a per-app basis. Does not retain either its keys or its values, since FIRApps may be deleted
+// without notifying their respective FIRStorage instances, and FIRStorage instances may be short-
+// lived.
++ (NSMapTable<FIRApp *, FIRFunctions *> *)instanceCache {
```

Comment:
@ryanwilson Please confirm this is ok from an API stability perspective

Reply:
This may result in surprising behavior since consumers don't tend to retain the main [`FIRFunctions functions`] instance. Thoughts?